File: src/main/java/sprykerFileCreator/model/definition/spryker/YvesFactoryDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class YvesFactoryDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.YVES_FACTORY;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}Factory";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Yves\\{{ bundleName }}";
    }

    @Override
    public String getMethodForReturnType() {
        return "getFactory()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {};

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Yves\\Kernel\\AbstractFactory";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/YvesFactoryDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class YvesFactoryDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.YVES_FACTORY
    override val namePattern: String
        get() = "{{ bundleName }}Factory"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Yves\\{{ bundleName }}"
    override val methodForReturnType: String
        get() = "getFactory()"
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Yves\\Kernel\\AbstractFactory"
}
